### PR TITLE
feat: init GTM Browser 2.0

### PR DESCRIPTION
**Pull request by yuhao900914**
# Amplitude Developer Docs PR

## Description

https://pr-888.d19s7xzcva2mw3.amplifyapp.com/data/sources/google-tag-manager-client/?h=gtm

Describe your changes. 
- added the warning for breaking changes
- added the breaking changes checklist
Screen Shot 2023-08-08 at 1 42 12 PM
- added the new features related to DES and re-ordered the sections to make them align with the template UI.

## Deadline

When do these changes need to be live on the site?

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [ ] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [ ] My documentation follows the style guidelines of this project.
- [ ] I previewed my documentation on a local server using `mkdocs serve`.
- [ ] Running `mkdocs serve` didn't generate any failures.
- [ ] I have performed a self-review of my own documentation.

@amplitude-dev-docs

### CODE REVIEW

File: docs/data/sources/google-tag-manager-client.md [hunk 1]
```diff
@@ -3,17 +3,45 @@ title: Google Tag Manager Web Template - Amplitude Analytics Browser SDK
...
+ 
+    While this updated template offers additional features and improvements,it may result in slightly different behavior that could potentially affect your existing analytics charts. However, we've made these changes configurable to maintain consistency. If you wish to retain the legacy behavior, it's essential to review the following [list of breaking changes](./#breaking-changes-checklist) and adjust your configuration accordingly.
+
 !!!warning
     Due to inherent limitations of GTM, certain features, such as plugins, are not well supported in this GTM template. You are still able to add plugins using the Custom HTML tag, but because of how the SDK gets loaded in GTM, this could lead to missing data.
 
 !!!note
     Ensure to consistently update your Amplitude GTM template to the latest version for an enhanced feature set, crucial bug fixes, and a significantly improved user experience.
 
+## Breaking changes checklist
```
Comment (yuhao900914) [inline docs/data/sources/google-tag-manager-client.md:22]: Let me know if you have a better idea for the checklist.

File: docs/data/sources/google-tag-manager-client.md [hunk 2]
```diff
@@ -3,17 +3,45 @@ title: Google Tag Manager Web Template - Amplitude Analytics Browser SDK
...
+
+### user agent parser
```
Comment (justin-fiedler) [inline docs/data/sources/google-tag-manager-client.md:24]: I feel this subheaders should be title case. e.g. "User agent parser" "Page view event_type and event_properties" etc

File: docs/data/sources/google-tag-manager-client.md [hunk 3]
```diff
@@ -3,17 +3,45 @@ title: Google Tag Manager Web Template - Amplitude Analytics Browser SDK
...
+
+???Breaking change "User Agent Parser"
+    | <div class="big-column">Before</div>  | Current | Effect Area | Solution |
+    | --- | --- | --- | --- |
+    | Using [client-side user agent parser](https://github.com/amplitude/ua-parser-js).  | Using server-side user agent parser which is offered by Amplitude ingestion endpoints. | <ul><li>Value of `event.os_name`, `event.os_version`, `event.device_model`, `event.device_manufacturer` and related properties.</li><li>The chart analytics</li></ul>| Enable client side user agent enrichment. [Read More](./#enable-client-side-user-agent-enrichment). |
```
Comment (justin-fiedler) [inline docs/data/sources/google-tag-manager-client.md:29]:
```suggestion
    | [Client-side user agent parsing](https://github.com/amplitude/ua-parser-js).  | Server-side user agent parsing by Amplitude ingestion endpoints. | <ul><li>Value of `event.os_name`, `event.os_version`, `event.device_model`, `event.device_manufacturer` and related properties.</li><li>The chart analytics</li></ul>| Enable client side user agent enrichment. [Read More](./#enable-client-side-user-agent-enrichment). |
```

File: docs/data/sources/google-tag-manager-client.md [hunk 4]
```diff
@@ -3,17 +3,45 @@ title: Google Tag Manager Web Template - Amplitude Analytics Browser SDK
...
+
+### page view event type and properties
+
+???Breaking change "Page View Tracking"
+    | <div class="big-column">Before</div>  | Current | Effect Area | Solution |
+    | --- | --- | --- | --- |
+    | <ul><li>event_type: `Page View`</li><li>event_properties: `page_location`, `page_path`, `page_title`. `page_url`</li></ul> | <ul><li>event_type:  `[Amplitude] Page Viewed`</li><li>event properties: `[Amplitude] Page Domain`, `[Amplitude] Page Location`, `[Amplitude] Page Path`, `[Amplitude] Page Title`, `[Amplitude] Page URL`</li></ul> | <ul><li>The chart analytic.</li><li>Unexpected tracking plans if you have Data.</li></ul>| <ul><li>Use the legacy event type and event properties of page view by checking `Use the legacy page view properties` checkbox.</li><li>Overwrite the display name in Data dashboard.</li></ul>|
```
Comment (justin-fiedler) [inline docs/data/sources/google-tag-manager-client.md:36]:
```suggestion
    | <ul><li>event_type: `Page View`</li><li>event_properties: `page_location`, `page_path`, `page_title`. `page_url`</li></ul> | <ul><li>event_type:  `[Amplitude] Page Viewed`</li><li>event properties: `[Amplitude] Page Domain`, `[Amplitude] Page Location`, `[Amplitude] Page Path`, `[Amplitude] Page Title`, `[Amplitude] Page URL`</li></ul> | <ul><li>The chart analytic.</li><li>Events may show as unexpected in tracking plans in Data.</li></ul>| <ul><li>Use the legacy event type and event properties of page view by checking `Use the legacy page view properties` checkbox.</li><li>Overwrite the display name in Data dashboard.</li></ul>|
```

File: docs/data/sources/google-tag-manager-client.md [hunk 5]
```diff
@@ -135,6 +149,57 @@ If the userId already available you can initialize the instance with a User ID.
...
+    | `Page View trigger` | `Page Loads` or `Only with Attribution changes` or a `Variable Configuration`.  The trigger of page view event. A variable configuration can be either build-in or customized that returns a function with a true or false return value. If the function returns true, then Page Views are tracked automatically, if it returns false then Page Views are not tracked. [More details](../../sdks/browser-2/#advanced-configuration-for-tracking-page-views). | `Page Loads` if enable page view tracking. |
+    | `Track history events automatically` | `All history changes` or `Only when page path changes`. Whether to track history events. This is for tracking page view on SPA. [More details](../../sdks/browser-2/#advanced-configuration-for-tracking-page-views). | `All history changes` |
+
+##### Sessions Tracking
+
+Check this box to enable sessions tracking. [More details](../../sdks/browser-2/#tracking-sessions).
+
+##### Form Interactions Tracking
+
+Check this box to enable form interactions tracking. [More details](../..//sdks/browser-2/#tracking-form-interactions).
```
Comment (justin-fiedler) [inline docs/data/sources/google-tag-manager-client.md:197]: nit: double slash. Also on line 201, 229, maybe others
```suggestion
Check this box to enable form interactions tracking. [More details](../../sdks/browser-2/#tracking-form-interactions).
```

**Review by justin-fiedler — COMMENTED** [review]
Overall looks good. I feel it might be better to pull "Effect area" and "Solution" out of the table and make it plain text under the header (as a summary) then the customer can expand the table if needed for more details.

File: docs/data/sources/google-tag-manager-client.md [hunk 6]
```diff
@@ -135,6 +149,57 @@ If the userId already available you can initialize the instance with a User ID.
...
 
+##### Enable client side user agent enrichment
+
+Starting with version xxxxx, as of Aug, 2023, we have upgraded this template to use the Amplitude Browser SDK 2.0 for data collection. In Amplitude Browser SDK 2.0, we have deprecated client-side user agent parsing in favor of server-side user agent parsing. [More details](https://github.com/amplitude/Amplitude-TypeScript/tree/v1.x/packages/plugin-user-agent-enrichment-browser).
+
+To avoid breaking changes in chart analytics, we provide the option to enable client-side parsing. If you are a new user. We highly recommend adopting use server-side parsing, the new enrichment strategy offers more accurate result.
+
+##### Default Event Tracking
+
+This section include:
```
Comment (Mercy811) [inline docs/data/sources/google-tag-manager-client.md:160]:
```suggestion
Check this checkbox to enable default event tracking and configure the following default tracking events
```

File: docs/data/sources/google-tag-manager-client.md [hunk 7]
```diff
@@ -135,6 +149,57 @@ If the userId already available you can initialize the instance with a User ID.
...
+
+This section include: 
+
+- Attribution Tracking
+- Page View Tracking
+- Sessions Tracking
+- Form Interaction Tracking
+- File downloads Tracking
+
+##### Attribution tracking
```
Comment (Mercy811) [inline docs/data/sources/google-tag-manager-client.md:168]: I notice that the header level is quite deep here but can we make this and the following headers to header 6 with ###### as this should be nested in default event tracking in UI

**Review by Mercy811 — COMMENTED** [review]
Thank you Alyssa for updating this 🙏 see nits

File: docs/data/sources/google-tag-manager-client.md (hunk 2 above)
Comment (kevinpagtakhan) [inline docs/data/sources/google-tag-manager-client.md:24]: same comment as before, i don't think this should be placed higher than the breaking changes for page view and cross subdomain attribution tracking

File: docs/data/sources/google-tag-manager-client.md [hunk 8]
```diff
@@ -135,6 +149,57 @@ If the userId already available you can initialize the instance with a User ID.
     |`transport`| `TransportType.XHR` or `TransportType.SendBeacon` or `TransportType.Fetch`. Set the transport type. | `TransportType.Fetch` |
 - `Select a **GTM variable** from the list`. It's necessary to return an object containing the key-value pairs you wish to use for instance configuration. Ensure that the keys are part of the available configurations.
 
+##### Enable client side user agent enrichment
+
+Starting with version xxxxx, as of Aug, 2023, we have upgraded this template to use the Amplitude Browser SDK 2.0 for data collection. In Amplitude Browser SDK 2.0, we have deprecated client-side user agent parsing in favor of server-side user agent parsing. [More details](https://github.com/amplitude/Amplitude-TypeScript/tree/v1.x/packages/plugin-user-agent-enrichment-browser).
+
+To avoid breaking changes in chart analytics, we provide the option to enable client-side parsing. If you are a new user. We highly recommend adopting use server-side parsing, the new enrichment strategy offers more accurate result.
```
[the remark this answers is not in the document]
Comment (yuhao900914) [reply docs/data/sources/google-tag-manager-client.md:156]: update to `If you are a new user, we highly recommend adopting server-side parsing by leaving this checkbox empty as the new enrichment strategy offers more accurate result.`

File: docs/data/sources/google-tag-manager-client.md [hunk 9]
```diff
@@ -135,6 +149,57 @@ If the userId already available you can initialize the instance with a User ID.
     |`transport`| `TransportType.XHR` or `TransportType.SendBeacon` or `TransportType.Fetch`. Set the transport type. | `TransportType.Fetch` |
 - `Select a **GTM variable** from the list`. It's necessary to return an object containing the key-value pairs you wish to use for instance configuration. Ensure that the keys are part of the available configurations.
 
+##### Enable client side user agent enrichment
+
+Starting with version xxxxx, as of Aug, 2023, we have upgraded this template to use the Amplitude Browser SDK 2.0 for data collection. In Amplitude Browser SDK 2.0, we have deprecated client-side user agent parsing in favor of server-side user agent parsing. [More details](https://github.com/amplitude/Amplitude-TypeScript/tree/v1.x/packages/plugin-user-agent-enrichment-browser).
```
[the remark this answers is not in the document]
Comment (yuhao900914) [reply docs/data/sources/google-tag-manager-client.md:154]: This is specific for the user agent enrichment. 
I agree that we should move the sha somewhere. But I don't think it's possible to create a variable in the template.
Will leave it there for now.

**Comment by yuhao900914** [comment]
> Overall looks good. I feel it might be better to pull "Effect area" and "Solution" out of the table and make it plain text under the header (as a summary) then the customer can expand the table if needed for more details. wdyt?
@justin-fiedler 
How do you like this layout?
Personally, I think it takes too much space and make the page too long. I prefer the previous layout. 
Screen Shot 2023-08-11 at 5 16 29 PM

**Comment by justin-fiedler** [comment]
> How do you like this layout?
> Personally, I think it takes too much space and make the page too long. I prefer the previous layout.

Ya I agree. It takes a lot of space with the sub headers. I think the table layout works well enough. Thanks

**Comment by yuhao900914** [comment]
> ```
> The new template changes the default page view events to include `[Amplitude]` prefixes. If you want to continue using the older page view events check `Use legacy page view properties`. See full details in table. 
> ```
@justin-fiedler  please check this. I feel this version is much better.
Screen Shot 2023-08-14 at 1 57 13 PM

File: docs/data/sources/google-tag-manager-client.md [hunk 10]
```diff
@@ -3,17 +3,51 @@ title: Google Tag Manager Web Template - Amplitude Analytics Browser SDK
...
+Traffic from one subdomain to another (ie analytics.amplitude.com to experiment.amplitude.com) is not tracked with no additional configuration. If you want to only want to exclude the attribution tracking on `location.hostname`, not other subdmaion, add the value of `location.hostname` in the exclude referral section. See full details in table. 
+
+???Breaking change "Attribution Tracking"
+    | <div class="big-column">Before</div>  | Current |
+    | --- | --- | --- | --- |
+    | Track attribution of all subdomains. | Excludes all subdomains of the same root domain as referrer | 
+
+### User agent parser
+
+The new template changes they way to parse the device related info which might effect the value of `event.os_name`, `event.os_version`, `event.device_model`, `event.device_manufacturer` and related properties. If you want to continue using the older page view events check `Use legacy page view properties`. See full details in table.
```
Comment (justin-fiedler) [inline docs/data/sources/google-tag-manager-client.md:44]: Use legacy page view properties?

File: docs/data/sources/google-tag-manager-client.md [hunk 11]
```diff
@@ -369,3 +340,34 @@ If you hardcode the value in your tag, the input will be forced into a string ty
...
+Traffic from one subdomain to another (ie analytics.amplitude.com to experiment.amplitude.com) is not tracked by default. If you want to exclude the attribution tracking on `location.hostname`, but not other subdomains, add the value of `location.hostname` in the exclude referral section. See full details in table. 
+
+???Breaking change "Attribution Tracking"
+    | <div class="big-column">Before</div>  | Current |
+    | --- | --- | --- | --- |
+    | Track attribution of all subdomains. | Excludes all subdomains of the same root domain as referrer | 
+
+### User agent parser
+
+The new template changes they way to parse the device related info which might effect the value of `event.os_name`, `event.os_version`, `event.device_model`, `event.device_manufacturer` and related properties. If you want to continue using the older way to parse user agent, check `Enable client side user agent enrichment `. See full details in table.
```
Comment (yuhao900914) [inline docs/data/sources/google-tag-manager-client.md:367]: `Enable client side user agent enrichment` -> `(Legacy) Enable client side user agent enrichment`

File: docs/data/sources/google-tag-manager-client.md [hunk 12]
```diff
@@ -83,27 +88,26 @@ A tag type allow you to specify what kind of action or event should be tracked i
 
 Cookies are generated at the initialization stage. For more information on managing cookies, please refer to our [cookie management details](../../sdks/typescript-browser/#cookie-management). It is recommended to initialize after obtaining cookie consent. As we support deferred initialization, any event tracking that takes place before the init command will still be captured and won't be lost.
 
-##### EU Data Residency
+##### Default Event Tracking
 
-For EU data residency, you must set up your project inside Amplitude EU and use the API key from Amplitude EU. You can configure the server zone by checking the checkbox **EU Data Residency** under **Tag Configuration** -> **Initialization** of the `init` tag. The initialization section only shows up when tag type is set to `init`. [More details](../../sdks/typescript-browser/#eu-data-residency).
+Check this checkbox to enable default event tracking and configure the following default tracking events
```
Comment (yuhao900914) [inline docs/data/sources/google-tag-manager-client.md:93]: `enable default event tracking` to `Track default Amplitude events`

**Comment by yuhao900914** [comment]
@AlecBoutin 
Since you update the name of the subheader or check box, please go over all of them and make it update to the template. 
Also I feel it will be better to create another pr to merge with this, otherwise, it's hard to compare the changes between yours and others.  I cannot approve this one either.

**Comment by yuhao900914** [comment]
@AlecBoutin 
Also, I think we'd better mention the changes of the apiKey and the instance name. 

and emphasize on all tags forr a particular api key should have the same instance name.

File: docs/data/sources/google-tag-manager-client.md [hunk 13]
```diff
@@ -41,7 +41,7 @@ Traffic from one subdomain to another (ie analytics.amplitude.com to experiment.
 
 ### User agent parser
 
-The new template changes they way to parse the device related info which might effect the value of `event.os_name`, `event.os_version`, `event.device_model`, `event.device_manufacturer` and related properties. If you want to continue using the older page view events check `Use legacy page view properties`. See full details in table. 
+The new template changes they way to parse the device related info which might effect the value of `event.os_name`, `event.os_version`, `event.device_model`, `event.device_manufacturer` and related properties. If you want to continue using the older way to parse user agent, check `Enable client side user agent enrichment `. See full details in table.
```
Comment (Mercy811) [inline docs/data/sources/google-tag-manager-client.md]:
```suggestion
The new template changes the way to parse the device related info which might effect the value of `event.os_name`, `event.os_version`, `event.device_model`, `event.device_manufacturer` and related properties. If you want to continue using the older way to parse user agent, check `Enable client side user agent enrichment `. See full details in table. 
```

**Pull request merged**